File: java/src/main/java/deltix/luminary/LiteralText.java
package deltix.luminary;

public class LiteralText extends Literal {
    private String value;

    public LiteralText(String value) {
        super(LiteralKind.TEXT);
        this.value = value;
    }

    public String getValue() {
        return value;
    }

    @Override
    public String toString() {
        return "\"" + value + "\"";
    }
}

File: csharp/main/LiteralText.cs
using System;

namespace Deltix.Luminary
{
	public sealed class LiteralText : Literal, ILiteral<String>
	{
		public LiteralText(String value)
			: base(LiteralKind.Text)
		{
			Value = value;
		}

		public String Value { get; }

		public override String ToString()
		{
			return $"\"{Value}\"";
		}
	}
}
